src/eccodes/grib_jasper_encoding.cc

```diff
@@ -215,7 +262,15 @@ int grib_jasper_encode(grib_context* c, j2k_encode_helper* helper)
         }
     }
 
-    ecc_jasper_initialise();
+    GRIB_MUTEX_INIT_ONCE(&once, &init);
+    GRIB_MUTEX_LOCK(&mutex);
+    jaserr = ecc_jasper_initialise();
+    GRIB_MUTEX_UNLOCK(&mutex);
+    if (jaserr) {
```

Comment:
Just noting I added this error check here as it seemed like it should exist in both encode and decode pathways.